src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -214,6 +214,7 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
                 var errorHandled = false;
                 var numberOfDeliveryAttempts = 0;
 
+                contextBag.Set(message);
```

Comment:
@SimonCropp FYI, for this to work the way you're intending, you'll have to also undo the optimization currently in place around the creation of the `ContextBag`. Currently, because we don't use the bag for anything, there is a single static instance that is reused for every message, saving having to allocate one per message.